### PR TITLE
chore(#621): update eslint to v8

### PR DESCRIPTION
**Pull request by m5r**
# Description

Update ESLint to v8 without migrating to the new configuration format

medic/cht-conf#621

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or integration tests where appropriate
- Backwards compatible: Works with existing data and configuration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by m5r** [comment]
@jkuester I was hoping to merge this to `main` after the previous PR but since Sonar won't play nice with my `TODO` comments I'm merging this to `621-update-non-breaking-deps` instead and will merge the previous PR to `main` once this gets merged

**Review by jkuester — APPROVED** [review]
How confident are we that this is not going to have a significant effect on the linting of partner configs?  Seems like it should be pretty much passive since we are not actually changing any linting rules.

If there is an unexpected problem, folks should be able to bypass the errors via setting the `--debug` flag when running `compile-app-settings`.

File: src/lib/compilation/package-lib.js
Comment (m5r) [reply src/lib/compilation/package-lib.js:96]: Yes exactly! `eslint-loader` is deprecated and the replacement `eslint-webpack-plugin` behaves differently with ESLint errors where it [treats errors as webpack warnings](https://github.com/webpack-contrib/eslint-webpack-plugin/blob/2.x/src/index.js#L150-L153) when `failOnError === false`
That's why this logic moved into the `warnings` block underneath

**Comment by m5r** [comment]
> How confident are we that this is not going to have a significant effect on the linting of partner configs? Seems like it should be pretty much passive since we are not actually changing any linting rules.

@jkuester I'm very confident as ESLint v8 still supports the config format we (and partner configs) are using, we have automated tests covering linting config code, and I ran the `compile-app-settings` command with this branch on a handful of the most recently updated partner configs without any issue.
And yeah, worst case scenario, if they do run into a linting error they can always bypass it with the `--debug` flag or by overriding rules in their own `.eslintrc` config

**Pull request merged**
